include/envoy/server/BUILD

```diff
@@ -172,6 +173,7 @@ envoy_cc_library(
     hdrs = ["transport_socket_config.h"],
     deps = [
         "//include/envoy/network:transport_socket_interface",
+        "//include/envoy/secret:secret_manager_interface",
```

Comment:
Nit: This isn't needed anymore.

Reply:
Removed

Reply:
secret_manager_interface is required here again.